Add advisory info on enabling dbs with ACM in helm chart

stevenGravy opened the pull request.

stevenGravy committed:
Add advisory info on enabling dbs with ACM

ptgott commented on docs/pages/deploy-a-cluster/helm-deployments/aws.mdx:
```diff
@@ -262,6 +262,15 @@ $ kubectl --namespace teleport create -f aws-issuer.yaml
 
 <Admonition type="warning" title="Warning">
+  
+  Using ACM through a AWS Load Balancer prevents the required traffic for Postgres or MongoDB through the 
```
Is there a more specific way to phrase "Using ACM through a AWS Load Balancer"? I'm not sure I understand this. Also, what does "required traffic" mean?

ptgott replied:
```suggestion
  Using ACM through a AWS Load Balancer prevents the required traffic for Postgres or MongoDB through Teleport's 
```
assuming this is correct—"the web port" can be more specific.

ptgott commented on docs/pages/deploy-a-cluster/helm-deployments/aws.mdx:
```diff
@@ -262,6 +262,15 @@ $ kubectl --namespace teleport create -f aws-issuer.yaml
 ## Step 4b/7. Configure Teleport to use ACM to handle TLS
 
 <Admonition type="warning" title="Warning">
+  
```
I think we can organize this Admonition a lot better. Right now, it seems like a collection of miscellaneous notes. 

The new text in this Admonition is more of a mitigation step that applies after you have already set up Teleport with ACM. Can we add that new text to the next step instead? I think a `Details` box would make sense for this.

ptgott commented on docs/pages/deploy-a-cluster/helm-deployments/aws.mdx:
```diff
@@ -262,6 +262,15 @@ $ kubectl --namespace teleport create -f aws-issuer.yaml
+  
+  Using ACM through a AWS Load Balancer prevents the required traffic for Postgres or MongoDB through the 
+  web port.  Add the listener variables with the `true` value that match your required database traffic types
+  to the values file in the next step.
```
```suggestion
  to the values file in the next step:
```

ptgott commented on docs/pages/deploy-a-cluster/helm-deployments/aws.mdx:
```diff
@@ -262,6 +262,15 @@ $ kubectl --namespace teleport create -f aws-issuer.yaml
 <Admonition type="warning" title="Warning">
+  
+  Using ACM through a AWS Load Balancer prevents the required traffic for Postgres or MongoDB through the 
+  web port.  Add the listener variables with the `true` value that match your required database traffic types
```
```suggestion
  web port. Add the following options, depending on whether you are running PostgreSQL or MongoDB,
```

ptgott commented:
@stevenGravy Following up to see if you still want to move this one forward

ptgott commented:
@stevenGravy Here is another follow-up nudge. Thanks!

stevenGravy committed:
verbiage change

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

ptgott commented:
@stevenGravy Just wondering what you thought of the comments I left in my earlier review. If this is fairly urgent and you'd like me to have a go at reorganizing this information, I'd be happy to.

stevenGravy committed:
verbiage correction

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
verbiage update

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

ptgott commented:
@stevenGravy Added a reorganization suggestion in this PR: https://github.com/gravitational/teleport/pull/19225

ptgott committed:
Reorganize advistory info in the AWS guide (#19225)

* Reorganize advistory info in the AWS guide

- Add all advisory information about the approach to choose (ACM vs.
  cert-manager) before the instructions for each approach. This way, the
  user can see all relevant information together before making the
  choice.

- To better express that the reader is making a choice between ACM and
  cert-manager, turn steps 4a and 4b into a `Tabs` component. To prevent
  ugliness caused by having a double-`Tabs` component in the ACM
  section, remove the `TabItem` re: using `--set`, since `values.yaml`
  files are cleaner, and use only the values file instructions.

* Mention using existing credentials

* Respond to PR feedback

alexfornuto commented on docs/pages/deploy-a-cluster/helm-deployments/aws.mdx:
```diff
@@ -116,22 +116,63 @@ Teleport will always look at the oldest version of a recording.
+internal certificate authority instead of Let's Encrypt or AWS Certificate
+Manager.
+
+<Notice type="warning">
+
+If you would like the Teleport Application Service and Database Service to
+function as expected, you should use the `cert-manager` approach unless there is
+a specific reason to use ACM.
+
+</Notice>
```
Does this notice belong in the previous section?

ptgott replied:
It does 🤦

alexfornuto replied:
https://www.youtube.com/watch?v=Ag1o3koTLWM

alexfornuto reviewed (approved):
1 nit, otherwise 👍

ptgott committed:
Respond to PR feedback
